### PR TITLE
Adds autorest to the common Dockerfile

### PR DESCRIPTION
**Pull request by krabhishek8260**
Adding autorest to the common Dockerfile instead of separate Dockerfiles and scripts for each language. 
Removes the duplicate code in the language specific scripts/Dockerfile which makes maintaining them easier.

### CODE REVIEW

**Comment by krabhishek8260** [comment]
@tg123 @brendanburns Review please

**Comment by krabhishek8260** [comment]
https://github.com/kubernetes-client/gen/pull/28

**Comment by krabhishek8260** [comment]
https://github.com/kubernetes-client/csharp/pull/36

File: openapi/Dockerfile [hunk 1]
```diff
@@ -32,6 +32,15 @@ RUN cd /source/swagger-codegen && \
     mvn install -DskipTests -Dmaven.test.skip=true -pl modules/swagger-codegen-maven-plugin -am && \
     cp -r /root/.m2/* /usr/share/maven/ref
 
+# Install Autorest
+RUN curl -sL https://deb.nodesource.com/setup_7.x | bash -
```
Comment (brendandburns) [inline openapi/Dockerfile]: don't we want to use 8.x? isn't 7.x unstable?

File: openapi/Dockerfile [hunk 2]
```diff
@@ -32,6 +32,15 @@ RUN cd /source/swagger-codegen && \
...
     cp -r /root/.m2/* /usr/share/maven/ref
 
+# Install Autorest
+RUN curl -sL https://deb.nodesource.com/setup_7.x | bash - 
+RUN apt-get update && apt-get -y install \
+    nodejs \
+    libunwind8-dev \
+    && rm -rf /var/lib/apt/lists/*
+
+RUN npm install -g autorest@preview
```
Comment (brendandburns) [inline openapi/Dockerfile]: Do we always want to install +@preview+? Perhaps we want to add a TODO here to revert to a more stable release when the features we need are there?

**Comment by brendandburns** [comment]
Small comments, but basically LGTM.

**Comment by krabhishek8260** [comment]
@brendanburns addressed the comments.

**Comment by brendandburns** [comment]
LGTM.

@mbohlool do you want to check this before I merge?

**Comment by mbohlool** [comment]
LGTM

**Comment by tg123** [comment]
LGTM 

better solution than #27 👍

**Pull request merged**